lib/ui/platform_dispatcher.dart

```diff
@@ -349,9 +351,20 @@ class PlatformDispatcher {
     return PointerDataPacket(data: data);
   }
 
-  /// Called by [_dispatchKeyData].
-  void _respondToKeyData(int responseId, bool handled)
-      native 'PlatformConfiguration_respondToKeyData';
+  static ChannelCallback _keyDataListener(KeyDataCallback onKeyData, Zone zone) =>
+    (ByteData? packet, PlatformMessageResponseCallback callback) {
+      _invoke1<KeyData>(
+        (KeyData keyData) {
+          final bool handled = onKeyData(keyData);
+          final Uint8List response = Uint8List.fromList(<int>[
+            if (handled) 1 else 0,
+          ]);
+          callback(ByteData.sublistView(response));
```

Comment:
nit:

```suggestion
          final Uint8List response = Uint8List(1);
          response[0] = handled ? 1 : 0;
          callback(response.buffer.asByteData());
```

I don't know if the compiler is clever enough to recognize that the list you're forming is always lenght of one or not, but if it doesn't this will result in bounds checking etc.